src/helpers/PromiseCollector.spec.ts

```diff
@@ -58,4 +58,14 @@ describe('PromiseCollector', () => {
       done();
     });
   });
+
+  it('swallows errors when removing promises with .finally', async() => {
+    // The "fail" case of this test is with a process exit code
```

Comment:
I had wondered the same, and found out something interesting: jest creates its own process so it can attach handlers, which makes it difficult/awkward to work around. I found some interesting looking suggestions on the jest issue, like https://github.com/facebook/jest/issues/5620#issuecomment-761879230 and https://github.com/facebook/jest/issues/5620#issuecomment-998716759 but wasn't sure we'd want to complicate the test suite like that. I could explore it more though and try to find something that feels like a good fit.